airflow/ui/src/components/DagVersionSelect.tsx

```diff
@@ -0,0 +1,101 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Field } from "@chakra-ui/react";
+import { useQueryClient } from "@tanstack/react-query";
+import type { OptionsOrGroups, GroupBase, SingleValue } from "chakra-react-select";
+import { AsyncSelect } from "chakra-react-select";
+import { useCallback } from "react";
+import { useSearchParams } from "react-router-dom";
+
+import { UseDagVersionServiceGetDagVersionsKeyFn } from "openapi/queries";
+import { DagVersionService } from "openapi/requests/services.gen";
+import type { DAGVersionCollectionResponse, DagVersionResponse } from "openapi/requests/types.gen";
+import { SearchParamsKeys } from "src/constants/searchParams";
+
+type Option = {
+  label: string;
+  value: string;
+};
```

Comment:
This is defined at multiple places, I'll open follow up PR to define this type once and for all.-

Reply:
PR here https://github.com/apache/airflow/pull/47119